TextEnrichment/MainForm.cs

```diff
@@ -0,0 +1,68 @@
+﻿using System.IO;
+using System.Windows.Forms;
+using TextEnrichment.Text;
+using TextEnrichment.Word;
+
+namespace TextEnrichment
+{
+    public partial class MainForm : Form
+    {
+        private readonly ISentencer sentencer;
+        private readonly IDocumentReader documentReader;
+        private readonly IDocumentWriter documentWriter;
+
+        public MainForm(ISentencer sentencer, IDocumentReader documentReader, IDocumentWriter documentWriter)
+        {
+            this.sentencer = sentencer;
+            this.documentReader = documentReader;
+            this.documentWriter = documentWriter;
+
+            InitializeComponent();
+        }
+
+        private void enrichBtn_Click(object sender, System.EventArgs e)
+        {
```

Comment:
I think that an IEnricher that responsible for the enrich method will be a better idea in terms of SoC (and you will not need to get ISentencer, IDocumentReader & IDocumentWriter but just IEnricher to this class).